File: client/src/app/shared/models/order.ts
import { Address } from './User';
import { DeliveryMethod } from './delivery';

export interface OrderToCreate {
  basketId: string;
  deliveryMethodId: number;
  shipToAddress: Address;
}

export interface Order {
  buyerEmail: string;
  orderItems: OrderItem[];
  deliveryMethod: DeliveryMethod;
  shipToAddress: Address;
  orderDate: string;
  orderStatus: number;
  subtotal: number;
  paymentIntentId: any;
  id: number;
}

export interface OrderItem {
  itemOrdered: ItemOrdered;
  price: number;
  quantity: number;
  id: number;
}

export interface ItemOrdered {
  productItemId: number;
  productName: string;
  pictureUrl: string;
}

export interface OrderToReturn {
  id: number;
  buyerEmail: string;
  orderItems: OrderItemToReturn[];
  deliveryMethod: string;
  shippingPrice: number;
  shipToAddress: Address;
  orderDate: string;
  orderStatus: string;
  subtotal: number;
}

export interface OrderItemToReturn {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

File: Core/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order()
        {
        }

        public Order(string buyerEmail, IReadOnlyList<OrderItem> orderItems, AddressOrder address,
        DeliveryMethod deliveryMethod, decimal subtotal, string paymentIntentId)
        {

            BuyerEmail = buyerEmail;
            OrderItems = orderItems;
            ShipToAddress = address;
            DeliveryMethod = deliveryMethod;
            Subtotal = subtotal;
            PaymentIntentId = paymentIntentId;
        }
        public string BuyerEmail { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public AddressOrder ShipToAddress { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

        public decimal Subtotal { get; set; }

        public string PaymentIntentId { get; set; }

        public decimal GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }
    }
}